From 7e8f2a4082198036475b47a0fc9df48a4f0e3521 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 17 Dec 2024 02:00:03 +0800
Subject: [PATCH 232/291] AOSCOS: MIPS: audit: clean up the last remnants of
 magic numbers

Follow commit 42f355ef59a2 ("audit: replace magic audit syscall class
numbers with macros") and replace all magic numbers in MIPS audit code.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/kernel/audit-n32.c    |  2 +-
 arch/mips/kernel/audit-native.c | 33 +++++++--------------------------
 2 files changed, 8 insertions(+), 27 deletions(-)

diff --git a/arch/mips/kernel/audit-n32.c b/arch/mips/kernel/audit-n32.c
index 8d114bf85425..76c95ff924f8 100644
--- a/arch/mips/kernel/audit-n32.c
+++ b/arch/mips/kernel/audit-n32.c
@@ -41,7 +41,7 @@ int audit_classify_syscall_n32(int abi, unsigned syscall)
 	case __NR_execve:
 		return AUDITSC_EXECVE;
 	default:
-		return 0;
+		return AUDITSC_NATIVE;
 	}
 }
 
diff --git a/arch/mips/kernel/audit-native.c b/arch/mips/kernel/audit-native.c
index 4a0fecd0e486..763f0e6b3636 100644
--- a/arch/mips/kernel/audit-native.c
+++ b/arch/mips/kernel/audit-native.c
@@ -41,45 +41,26 @@ int audit_classify_arch(int arch)
 
 int audit_classify_syscall(int abi, unsigned syscall)
 {
-	int res;
-
 	switch (syscall) {
 	case __NR_open:
-		res = AUDITSC_OPEN;
-		break;
-
+		return AUDITSC_OPEN;
 	case __NR_openat:
-		res = AUDITSC_OPEN;
-		break;
-
+		return AUDITSC_OPEN;
 #ifdef __NR_socketcall		/* Only exists on O32 */
 	case __NR_socketcall:
-		res = AUDITSC_SOCKETCALL;
-		break;
+		return AUDITSC_SOCKETCALL;
 #endif
 	case __NR_execve:
-		res = AUDITSC_EXECVE;
-		break;
+		return AUDITSC_EXECVE;
 	default:
 #ifdef CONFIG_AUDITSYSCALL_O32
-		res = audit_classify_syscall_o32(abi, syscall);
-		if (res)
-			break;
+		return audit_classify_syscall_o32(abi, syscall);
 #endif
 #ifdef CONFIG_AUDITSYSCALL_N32
-		res = audit_classify_syscall_n32(abi, syscall);
-		if (res)
-			break;
+		return audit_classify_syscall_n32(abi, syscall);
 #endif
-		if (abi == AUDIT_ARCH_MIPS || abi == AUDIT_ARCH_MIPSEL)
-			res = 1;
-		else if (abi == AUDIT_ARCH_MIPS64 || abi == AUDIT_ARCH_MIPSEL64)
-			res = 0;
-		else if (abi == AUDIT_ARCH_MIPS64N32 || abi == AUDIT_ARCH_MIPSEL64N32)
-			res = 6;
+		return AUDITSC_NATIVE;
 	}
-
-	return res;
 }
 
 static int __init audit_classes_init(void)
-- 
2.48.1

